toolkit/quickrebuild_autoconfig.mk

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Contains:
+#	- build optimization flag auto configuration
+
+######## QUICKREBUILD AUTO CONFIGURE ########
+
+# The QUICKREBUILD* flags are special flags that will try to build the toolchain and packages as quickly as possible. They will
+# automatically set REBUILD_TOOLS, USE_CCACHE, REBUILD_TOOLCHAIN, DELTA_BUILD, INCREMENTAL_TOOLCHAIN, and ALLOW_TOOLCHAIN_DOWNLOAD_FAIL to 'y'.
+QUICKREBUILD           ?= n
+QUICKREBUILD_TOOLCHAIN ?= n
+QUICKREBUILD_PACKAGES  ?= n
+
+ifeq ($(QUICKREBUILD),y)
+QUICKREBUILD_TOOLCHAIN = y
+QUICKREBUILD_PACKAGES  = y
+endif
+
+######## QUICKREBUILD TOOLCHAIN ########
+
+ifeq ($(QUICKREBUILD_TOOLCHAIN),y)
+# If any of these are already set to 'n', report an error
+ifneq ($(filter n,$(REBUILD_TOOLCHAIN)),)
+$(error QUICKREBUILD cannot be used with REBUILD_TOOLCHAIN explicitly set to 'n')
+endif
+ifneq ($(filter n,$(INCREMENTAL_TOOLCHAIN)),)
+$(error QUICKREBUILD cannot be used with INCREMENTAL_TOOLCHAIN explicitly set to 'n')
+endif
+ifneq ($(filter n,$(ALLOW_TOOLCHAIN_DOWNLOAD_FAIL)),)
+$(error QUICKREBUILD cannot be used with ALLOW_TOOLCHAIN_DOWNLOAD_FAIL explicitly set to 'n')
+endif
+
+# Don't care if REBUILD_TOOLS is set or not, doesn't matter to the quickbuild.
+
+REBUILD_TOOLCHAIN             = y
+INCREMENTAL_TOOLCHAIN         = y
+ALLOW_TOOLCHAIN_DOWNLOAD_FAIL = y
+REBUILD_TOOLS                 ?= y
```

Comment:
Why the `?=` in this case, if it's not set anywhere earlier?

Reply:
The behavior of the quickbuild flags isn't truly affected by REBUILD_TOOLS=y/n (hence why it isn't error checked), but I'm trying to make this the start of a 'works by default' way of interacting with the tools. So why not just turn that on by default for people, one less flag to worry about.

Reply:
I'll update comments for REBUILD_TOOLS and USE_CCACHE since they are both similar.

Reply:
For the logic: I feel like we should allow users to configure the tools to their needs where possible. In this case REBUILD_TOOLS=n won't stop the quick rebuild flow from working so I don't need to error check or override the user.

Make has some interesting precedence rules. While command line will always override the makefile, `?=` works as expected for exported environment variables (which I know some people use).
i.e.:
```makefile
FOO ?= baz
$(error echo $(FOO))
```
```bash
make
# baz

make FOO=bar
#bar

export FOO=fizz
make
#fizz
```
But using `FOO = baz` the last command would return `baz` rather than `fizz`.

Reply:
Scary stuff. Carry on then.:)